hooks/tk-multi-publish2/basic/nuke_update_flame_clip.py

```diff
@@ -372,82 +433,96 @@ def _update_flame_clip(self, item):
+                item.properties["sequence_paths"][0]
             )
 
-        # now when we have the real min/max frame, we can apply a proper
-        # sequence marker for the Flame xml. Note that we cannot use the normal
-        # FORMAT: token in the template system, because the Flame frame format
-        # is not totally "abstract" (e.g. %04d, ####, etc) but contains the
-        # frame ranges themselves.
-        #
-        # the format spec is something like "04"
-        sequence_key = publish_template.keys["SEQ"]
+            if not publish_path_flame:
+                raise Exception(
+                    "Couldn't extract min and max frame from the published "
+                    "sequence! Will not update Flame clip xml."
+                )
+        else:
+            # each publish task is connected to a nuke write node instance. this
+            # value was populated via the collector and verified during accept()
+            write_node = item.properties["sg_writenode"]
+
+            # get the fields from the work file
+            render_path = write_node_app.get_node_render_path(write_node)
+            render_template = write_node_app.get_node_render_template(write_node)
+            render_path_fields = render_template.get_fields(render_path)
+            publish_template = write_node_app.get_node_publish_template(write_node)
+
+            # set up the sequence token to be Flame friendly
+            # e.g. mi001_scene_output_v001.[0100-0150].dpx
+            # note - we cannot take the frame ranges from the write node -
+            # because those values indicate the intended frame range rather
+            # than the rendered frame range! In order for Flame to pick up
+            # the media properly, it needs to contain the actual frame data
+
+            # get all paths for all frames and all eyes
+            paths = self.parent.sgtk.paths_from_template(
+                publish_template,
+                render_path_fields,
+                skip_keys=["SEQ", "eye"]
+            )
 
-        # now compose the format string, eg. [%04d-%04d]
-        format_str = "[%%%sd-%%%sd]" % (
-            sequence_key.format_spec,
-            sequence_key.format_spec
-        )
+            # for each of them, extract the frame number. Track the min and the max.
+            # TODO: would be nice to have a convenience method in core for this.
+            min_frame = None
+            max_frame = None
+            for path in paths:
+                fields = publish_template.get_fields(path)
+                frame_number = fields["SEQ"]
+                if min_frame is None or frame_number < min_frame:
+                    min_frame = frame_number
+                if max_frame is None or frame_number > max_frame:
+                    max_frame = frame_number
+
+            # ensure we have a min/max frame
+            if min_frame is None or max_frame is None:
```

Comment:
This would only be the case if there were no frames, right?

Reply:
At a glance, I think you're correct. However, I'm not going to tinker with any of this old logic other than what I've already had to do.